### ISSUE TITLE
Concept name of record reverts to previous name during editing

### ISSUE THREAD

**Issue by SarahRDBingo** [mbari-media-management/vars-annotation#129]
when creating a new record or copying a previous record and changing the concept name to a new taxa, the taxon will update to the new name, but then revert back to the previous name or "object" after adding in details such as "upon" and "photo-reference." The concept name change happens when adding some detail associations, but not always. The name is more likely to change when details are edited in the associations pane. After associations are added and edited, the undesired concept name change can be undone by using the back button or cmd+z function.

**Comment by SarahRDBingo** [comment]
I'm hoping this is a derby issue that will be fixed if we transfer over to a PostgreSQL database

**Comment by hohonuuli** [comment]
@SarahRDBingo That's something we aren't experiencing at MBARI, so yes, it's likely related to Derby's performance. I worked on the Postgres port a few weeks ago. It still needs some more testing and then a couple of bits or code to transfer your data from derby into Postgres.

**Comment by hohonuuli** [comment]
@SarahRDBingo I *think* I've got the M3/VARS stack running against postgres now. As time permits, I'll copy your database into postgres and run some tests.

**Comment by SarahRDBingo** [comment]
That is fantastic news. Send us a message when we need to restart docker.
It would be great if we could start running it next week.

-Sarah

**Comment by hohonuuli** [comment]
> It would be great if we could start running it next week.

I'll see what I can do. It makes me nervous as I haven't done extensive testing. There could be speed bumps!

**Comment by hohonuuli** [comment]
@SarahRDBingo  Support of Postgres@DARC is going into the `develop` branch of [DARC/](https://github.com/DARC-UHM/m3-quickstart)

**Comment by SarahRDBingo** [comment]
@hohonuuli after some use, the issue of the concept name changing after adding/editing details to the record still exists, although it is not as apparent/prevalent as it used to be. We have also changed our SOP to only register videos with 2-hour runtime intervals or less. Maybe the behaviour is related to the number of records associated with a given video sequence?

**Comment by hohonuuli** [comment]
> Therefore, I believe that this issue should be moved to top priority to resolve as soon as possible.

@MeaganPutts OK.As a first look, can you zip up the log files in `$HOME/.vars/logs` and send them to me? That might give me some pointers to the problem.

**Comment by hohonuuli** [comment]
@MeaganPutts Also, which version of VARS are you running?

**Comment by MeaganPutts** [comment]
I last updated VARS April 6th. I see that there is a newer version available, so I will download that and let you know how it goes with this issue using the update.

[logs_06022022.zip](https://github.com/mbari-media-management/vars-annotation/files/8826709/logs_06022022.zip)

**Comment by hohonuuli** [comment]
I don't see this issue in the logs at all. Is this occurring on all your workstations? Can you go to Apple > About this Mac and post the os version, model year and memory here. (no serial number please)

**Comment by hohonuuli** [comment]
Also, I added a new VARS release that turns the amount of info way down. On your really ancient machines that might make a difference. It's at https://github.com/mbari-media-management/vars-annotation/releases/tag/1.2.2

**Comment by hohonuuli** [comment]
@MeaganPutts After typing the new name are you hitting [enter]? (Just to clarify that you should be ... )

**Comment by SarahRDBingo** [comment]
@hohonuuli The issue occurs on all our computers, including my new laptop running 16gb memory and OS Monterey version 12.2.1. The desired concept name is registered in the record after hitting [enter]. The problem occurs after the name is entered, the concept name changes to something else as details are added or edited on the record. Often the concept name becomes a repeat of the most recent record, or a previous name. The error can be reversed by clicking the back button, which leads me to suspect that the name change is not an issue of the right name not being saved correctly. Where do you want me to put the source code of version 1.2.2?

**Comment by hohonuuli** [comment]
@SarahRDBingo OK, that helps me narrow down possibilities. Does this occur when there are only a few annotations on a video, or only when there are lots of annotations? Can you point me at a video that you've seen this issue on?

**Comment by hohonuuli** [comment]
We may have to schedule a screen sharing session next week as I can't run VARS against your setup due to UofH's firewall.

**Comment by SarahRDBingo** [comment]
@hohonuuli I think that's a great idea. What is your schedule like on Monday or Tuesday. Also, there was no .dmg package in the 1.2.2 release page. Should I just copy the contents of the source code zip over to wherever the app lives in our directory?

**Comment by hohonuuli** [comment]
@SarahRDBingo OK DMG is now included in the release.

**Comment by hohonuuli** [comment]
@SarahRDBingo Monday is pretty open. So anytime (9-5 pacific) works for me.

**Comment by SarahRDBingo** [comment]
Can we do Tuesday morning instead? Meagan has a function to attend on Monday and I want her to be present.

**Comment by hohonuuli** [comment]
I have meetings from 9-10AM Pacific and 3:15-4:15PM Pacific on Tuesday. I'm free otherwise. Let me know a time that works for you and Meagan.

**Comment by SarahRDBingo** [comment]
Hi Brian,
Can we try for 11am PST (8AM HST) tomorrow morning?

**Comment by hohonuuli** [comment]
Log output from vars-annotation during interactive session when user created `object`, renamed it to `Bathypathes sp` and then it somehow changed to `Neomorphaster forcipatus`:

```text
2022-06-08 19:19:20 [JavaFX Application Thread       ] DEBUG org.mbari.vars.ui.AppController                     org.mbari.vars.ui.commands.CreateAnnotationFromConceptCmd@177a71e2
2022-06-08 19:19:20 [org.mbari.vars.ui.commands.CommandManager] DEBUG org.mbari.vars.ui.commands.CommandManager           Executing Command: DO - Create Annotation using object
2022-06-08 19:19:20 [org.mbari.vars.ui.commands.CommandManager] DEBUG org.mbari.vars.ui.AppController                     AnnotationsAddedEvent{source=null, refs=[Annotation{observationUuid=1d9e230f-46e1-4a60-976b-d55efe75ce1e, concept='object', elapsedTime=PT1H55M26.041S, numOfAssociations=0, numOfImages=0}]}
2022-06-08 19:19:20 [org.mbari.vars.ui.commands.CommandManager] DEBUG org.mbari.vars.ui.AppController                     AnnotationsSelectedEvent{source=null, refs=[Annotation{observationUuid=1d9e230f-46e1-4a60-976b-d55efe75ce1e, concept='object', elapsedTime=PT1H55M26.041S, numOfAssociations=0, numOfImages=0}]}
2022-06-08 19:19:20 [JavaFX Application Thread       ] DEBUG org.mbari.vars.ui.AppController                     AnnotationsSelectedEvent{source=org.mbari.vars.ui.javafx.annotable.AnnotationTableController@7dfa6a82, refs=[]}
2022-06-08 19:19:20 [JavaFX Application Thread       ] DEBUG org.mbari.vars.ui.AppController                     AnnotationsSelectedEvent{source=org.mbari.vars.ui.javafx.annotable.AnnotationTableController@7dfa6a82, refs=[Annotation{observationUuid=1d9e230f-46e1-4a60-976b-d55efe75ce1e, concept='object', elapsedTime=PT1H55M26.041S, numOfAssociations=0, numOfImages=0}]}
2022-06-08 19:19:21 [JavaFX Application Thread       ] DEBUG o.m.v.ui.javafx.shared.FilteredComboBoxDecorator    Handling KeyCode = B
2022-06-08 19:19:21 [JavaFX Application Thread       ] DEBUG o.m.v.ui.javafx.shared.FilteredComboBoxDecorator    Handling KeyCode = A
2022-06-08 19:19:21 [JavaFX Application Thread       ] DEBUG o.m.v.ui.javafx.shared.FilteredComboBoxDecorator    Handling KeyCode = T
2022-06-08 19:19:21 [JavaFX Application Thread       ] DEBUG o.m.v.ui.javafx.shared.FilteredComboBoxDecorator    Handling KeyCode = H
2022-06-08 19:19:23 [JavaFX Application Thread       ] DEBUG o.m.v.ui.javafx.shared.FilteredComboBoxDecorator    Handling KeyCode = Y
2022-06-08 19:19:23 [JavaFX Application Thread       ] DEBUG o.m.v.ui.javafx.shared.FilteredComboBoxDecorator    Handling KeyCode = P
2022-06-08 19:19:23 [JavaFX Application Thread       ] DEBUG o.m.v.ui.javafx.shared.FilteredComboBoxDecorator    Handling KeyCode = A
2022-06-08 19:19:23 [JavaFX Application Thread       ] DEBUG o.m.v.ui.javafx.shared.FilteredComboBoxDecorator    Handling KeyCode = T
2022-06-08 19:19:23 [JavaFX Application Thread       ] DEBUG o.m.v.ui.javafx.shared.FilteredComboBoxDecorator    Handling KeyCode = H
2022-06-08 19:19:24 [JavaFX Application Thread       ] DEBUG o.m.v.ui.javafx.shared.FilteredComboBoxDecorator    Handling KeyCode = E
2022-06-08 19:19:24 [JavaFX Application Thread       ] DEBUG o.m.v.ui.javafx.shared.FilteredComboBoxDecorator    Handling KeyCode = R
2022-06-08 19:19:24 [JavaFX Application Thread       ] DEBUG o.m.v.ui.javafx.shared.FilteredComboBoxDecorator    Handling KeyCode = BACK_SPACE
2022-06-08 19:19:24 [JavaFX Application Thread       ] DEBUG o.m.v.ui.javafx.shared.FilteredComboBoxDecorator    Handling KeyCode = S
2022-06-08 19:19:25 [JavaFX Application Thread       ] DEBUG o.m.v.ui.javafx.shared.FilteredComboBoxDecorator    Handling KeyCode = S
2022-06-08 19:19:26 [JavaFX Application Thread       ] DEBUG o.m.v.ui.javafx.shared.FilteredComboBoxDecorator    Handling KeyCode = P
2022-06-08 19:19:27 [JavaFX Application Thread       ] DEBUG org.mbari.vars.ui.AppController                     UpdateAnnotationCmd{oldAnnotation=Annotation{observationUuid=1d9e230f-46e1-4a60-976b-d55efe75ce1e, concept='object', elapsedTime=PT1H55M26.041S, numOfAssociations=0, numOfImages=0}, newAnnotation=Annotation{observationUuid=1d9e230f-46e1-4a60-976b-d55efe75ce1e, concept='Bathypathes sp', elapsedTime=PT1H55M26.041S, numOfAssociations=0, numOfImages=0}}
2022-06-08 19:19:27 [org.mbari.vars.ui.commands.CommandManager] DEBUG org.mbari.vars.ui.commands.CommandManager           Executing Command: DO - Update annotation
2022-06-08 19:19:27 [OkHttp [http://hurlstor.soest.hawaii.edu:8082/.](http://hurlstor.soest.hawaii.edu:8082/)..] DEBUG org.mbari.vars.ui.AppController                     AnnotationsChangedEvent{source=null, refs=[Annotation{observationUuid=1d9e230f-46e1-4a60-976b-d55efe75ce1e, concept='Bathypathes sp', elapsedTime=PT1H55M26.041S, numOfAssociations=0, numOfImages=0}]}
2022-06-08 19:19:27 [JavaFX Application Thread       ] DEBUG org.mbari.vars.ui.AppController                     AnnotationsSelectedEvent{source=org.mbari.vars.ui.javafx.annotable.AnnotationTableController@7dfa6a82, refs=[Annotation{observationUuid=90411b5a-daef-4cfe-7165-ce5afd75ce1e, concept='Neomorphaster forcipatus', elapsedTime=PT1H54M57.898S, numOfAssociations=4, numOfImages=0}]}
2022-06-08 19:19:27 [JavaFX Application Thread       ] DEBUG org.mbari.vars.ui.AppController                     AnnotationsSelectedEvent{source=null, refs=[Annotation{observationUuid=1d9e230f-46e1-4a60-976b-d55efe75ce1e, concept='Bathypathes sp', elapsedTime=PT1H55M26.041S, numOfAssociations=0, numOfImages=0}]}
2022-06-08 19:19:27 [JavaFX Application Thread       ] DEBUG org.mbari.vars.ui.AppController                     AnnotationsSelectedEvent{source=org.mbari.vars.ui.javafx.annotable.AnnotationTableController@7dfa6a82, refs=[Annotation{observationUuid=1d9e230f-46e1-4a60-976b-d55efe75ce1e, concept='Bathypathes sp', elapsedTime=PT1H55M26.041S, numOfAssociations=0, numOfImages=0}]}
2022-06-08 19:19:27 [JavaFX Application Thread       ] DEBUG org.mbari.vars.ui.AppController                     AnnotationsSelectedEvent{source=org.mbari.vars.ui.javafx.annotable.AnnotationTableController@7dfa6a82, refs=[Annotation{observationUuid=1d9e230f-46e1-4a60-976b-d55efe75ce1e, concept='Bathypathes sp', elapsedTime=PT1H55M26.041S, numOfAssociations=0, numOfImages=0}]}
2022-06-08 19:19:29 [JavaFX Application Thread       ] DEBUG org.mbari.vars.ui.AppController                     org.mbari.vars.ui.commands.CreateAssociationsCmd@23763c37
2022-06-08 19:19:29 [org.mbari.vars.ui.commands.CommandManager] DEBUG org.mbari.vars.ui.commands.CommandManager           Executing Command: DO - Add Association: s1 | bed | nil
2022-06-08 19:19:29 [OkHttp [http://hurlstor.soest.hawaii.edu:8082/.](http://hurlstor.soest.hawaii.edu:8082/)..] DEBUG org.mbari.vars.ui.AppController                     AnnotationsChangedEvent{source=null, refs=[Annotation{observationUuid=1d9e230f-46e1-4a60-976b-d55efe75ce1e, concept='Bathypathes sp', elapsedTime=PT1H55M26.041S, numOfAssociations=1, numOfImages=0}]}
2022-06-08 19:19:29 [OkHttp [http://hurlstor.soest.hawaii.edu:8082/.](http://hurlstor.soest.hawaii.edu:8082/)..] DEBUG org.mbari.vars.ui.AppController                     AnnotationsSelectedEvent{source=org.mbari.vars.ui.javafx.annotable.AnnotationTableController@7dfa6a82, refs=[Annotation{observationUuid=90411b5a-daef-4cfe-7165-ce5afd75ce1e, concept='Neomorphaster forcipatus', elapsedTime=PT1H54M57.898S, numOfAssociations=4, numOfImages=0}]}
2022-06-08 19:19:29 [JavaFX Application Thread       ] DEBUG org.mbari.vars.ui.AppController                     AnnotationsSelectedEvent{source=null, refs=[Annotation{observationUuid=1d9e230f-46e1-4a60-976b-d55efe75ce1e, concept='Bathypathes sp', elapsedTime=PT1H55M26.041S, numOfAssociations=1, numOfImages=0}]}
2022-06-08 19:19:29 [JavaFX Application Thread       ] DEBUG org.mbari.vars.ui.AppController                     AnnotationsSelectedEvent{source=org.mbari.vars.ui.javafx.annotable.AnnotationTableController@7dfa6a82, refs=[Annotation{observationUuid=1d9e230f-46e1-4a60-976b-d55efe75ce1e, concept='Bathypathes sp', elapsedTime=PT1H55M26.041S, numOfAssociations=1, numOfImages=0}]}
2022-06-08 19:19:29 [JavaFX Application Thread       ] DEBUG org.mbari.vars.ui.AppController                     AnnotationsSelectedEvent{source=org.mbari.vars.ui.javafx.annotable.AnnotationTableController@7dfa6a82, refs=[Annotation{observationUuid=1d9e230f-46e1-4a60-976b-d55efe75ce1e, concept='Bathypathes sp', elapsedTime=PT1H55M26.041S, numOfAssociations=1, numOfImages=0}]}
2022-06-08 19:19:31 [JavaFX Application Thread       ] DEBUG org.mbari.vars.ui.AppController                     org.mbari.vars.ui.commands.CreateAssociationsCmd@3cfbef44
2022-06-08 19:19:31 [org.mbari.vars.ui.commands.CommandManager] DEBUG org.mbari.vars.ui.commands.CommandManager           Executing Command: DO - Add Association: s2 | sed | nil
2022-06-08 19:19:31 [OkHttp [http://hurlstor.soest.hawaii.edu:8082/.](http://hurlstor.soest.hawaii.edu:8082/)..] DEBUG org.mbari.vars.ui.AppController                     AnnotationsChangedEvent{source=null, refs=[Annotation{observationUuid=1d9e230f-46e1-4a60-976b-d55efe75ce1e, concept='Bathypathes sp', elapsedTime=PT1H55M26.041S, numOfAssociations=2, numOfImages=0}]}
2022-06-08 19:19:31 [OkHttp [http://hurlstor.soest.hawaii.edu:8082/.](http://hurlstor.soest.hawaii.edu:8082/)..] DEBUG org.mbari.vars.ui.AppController                     AnnotationsSelectedEvent{source=org.mbari.vars.ui.javafx.annotable.AnnotationTableController@7dfa6a82, refs=[Annotation{observationUuid=90411b5a-daef-4cfe-7165-ce5afd75ce1e, concept='Neomorphaster forcipatus', elapsedTime=PT1H54M57.898S, numOfAssociations=4, numOfImages=0}]}
2022-06-08 19:19:31 [JavaFX Application Thread       ] DEBUG org.mbari.vars.ui.AppController                     AnnotationsSelectedEvent{source=null, refs=[Annotation{observationUuid=1d9e230f-46e1-4a60-976b-d55efe75ce1e, concept='Bathypathes sp', elapsedTime=PT1H55M26.041S, numOfAssociations=2, numOfImages=0}]}
2022-06-08 19:19:31 [JavaFX Application Thread       ] DEBUG org.mbari.vars.ui.AppController                     AnnotationsSelectedEvent{source=org.mbari.vars.ui.javafx.annotable.AnnotationTableController@7dfa6a82, refs=[Annotation{observationUuid=1d9e230f-46e1-4a60-976b-d55efe75ce1e, concept='Bathypathes sp', elapsedTime=PT1H55M26.041S, numOfAssociations=2, numOfImages=0}]}
2022-06-08 19:19:31 [JavaFX Application Thread       ] DEBUG org.mbari.vars.ui.AppController                     AnnotationsSelectedEvent{source=org.mbari.vars.ui.javafx.annotable.AnnotationTableController@7dfa6a82, refs=[Annotation{observationUuid=1d9e230f-46e1-4a60-976b-d55efe75ce1e, concept='Bathypathes sp', elapsedTime=PT1H55M26.041S, numOfAssociations=2, numOfImages=0}]}
2022-06-08 19:19:32 [JavaFX Application Thread       ] DEBUG org.mbari.vars.ui.AppController                     org.mbari.vars.ui.commands.CreateAssociationsCmd@6ab9a490
2022-06-08 19:19:32 [org.mbari.vars.ui.commands.CommandManager] DEBUG org.mbari.vars.ui.commands.CommandManager           Executing Command: DO - Add Association: size | 10-30 cm | nil
2022-06-08 19:19:32 [OkHttp [http://hurlstor.soest.hawaii.edu:8082/.](http://hurlstor.soest.hawaii.edu:8082/)..] DEBUG org.mbari.vars.ui.AppController                     AnnotationsChangedEvent{source=null, refs=[Annotation{observationUuid=1d9e230f-46e1-4a60-976b-d55efe75ce1e, concept='Bathypathes sp', elapsedTime=PT1H55M26.041S, numOfAssociations=3, numOfImages=0}]}
2022-06-08 19:19:32 [OkHttp [http://hurlstor.soest.hawaii.edu:8082/.](http://hurlstor.soest.hawaii.edu:8082/)..] DEBUG org.mbari.vars.ui.AppController                     AnnotationsSelectedEvent{source=org.mbari.vars.ui.javafx.annotable.AnnotationTableController@7dfa6a82, refs=[Annotation{observationUuid=90411b5a-daef-4cfe-7165-ce5afd75ce1e, concept='Neomorphaster forcipatus', elapsedTime=PT1H54M57.898S, numOfAssociations=4, numOfImages=0}]}
2022-06-08 19:19:32 [JavaFX Application Thread       ] DEBUG org.mbari.vars.ui.AppController                     AnnotationsSelectedEvent{source=null, refs=[Annotation{observationUuid=1d9e230f-46e1-4a60-976b-d55efe75ce1e, concept='Bathypathes sp', elapsedTime=PT1H55M26.041S, numOfAssociations=3, numOfImages=0}]}
2022-06-08 19:19:32 [JavaFX Application Thread       ] DEBUG org.mbari.vars.ui.AppController                     AnnotationsSelectedEvent{source=org.mbari.vars.ui.javafx.annotable.AnnotationTableController@7dfa6a82, refs=[Annotation{observationUuid=1d9e230f-46e1-4a60-976b-d55efe75ce1e, concept='Bathypathes sp', elapsedTime=PT1H55M26.041S, numOfAssociations=3, numOfImages=0}]}
2022-06-08 19:19:32 [JavaFX Application Thread       ] DEBUG org.mbari.vars.ui.AppController                     AnnotationsSelectedEvent{source=org.mbari.vars.ui.javafx.annotable.AnnotationTableController@7dfa6a82, refs=[Annotation{observationUuid=1d9e230f-46e1-4a60-976b-d55efe75ce1e, concept='Bathypathes sp', elapsedTime=PT1H55M26.041S, numOfAssociations=3, numOfImages=0}]}
2022-06-08 19:19:35 [JavaFX Application Thread       ] DEBUG o.m.v.ui.javafx.shared.FilteredComboBoxDecorator    Handling KeyCode = ENTER
2022-06-08 19:19:35 [JavaFX Application Thread       ] DEBUG org.mbari.vars.ui.javafx.buttons.UponBC             Select upon substrate of bed
2022-06-08 19:19:35 [JavaFX Application Thread       ] DEBUG org.mbari.vars.ui.AppController                     org.mbari.vars.ui.commands.CreateAssociationsCmd@17a5a66a
2022-06-08 19:19:35 [org.mbari.vars.ui.commands.CommandManager] DEBUG org.mbari.vars.ui.commands.CommandManager           Executing Command: DO - Add Association: upon | bed | nil
2022-06-08 19:19:35 [OkHttp [http://hurlstor.soest.hawaii.edu:8082/.](http://hurlstor.soest.hawaii.edu:8082/)..] DEBUG org.mbari.vars.ui.AppController                     AnnotationsChangedEvent{source=null, refs=[Annotation{observationUuid=1d9e230f-46e1-4a60-976b-d55efe75ce1e, concept='Bathypathes sp', elapsedTime=PT1H55M26.041S, numOfAssociations=4, numOfImages=0}]}
2022-06-08 19:19:35 [JavaFX Application Thread       ] DEBUG org.mbari.vars.ui.AppController                     AnnotationsSelectedEvent{source=org.mbari.vars.ui.javafx.annotable.AnnotationTableController@7dfa6a82, refs=[Annotation{observationUuid=90411b5a-daef-4cfe-7165-ce5afd75ce1e, concept='Neomorphaster forcipatus', elapsedTime=PT1H54M57.898S, numOfAssociations=4, numOfImages=0}]}
2022-06-08 19:19:35 [JavaFX Application Thread       ] DEBUG org.mbari.vars.ui.AppController                     AnnotationsSelectedEvent{source=null, refs=[Annotation{observationUuid=1d9e230f-46e1-4a60-976b-d55efe75ce1e, concept='Bathypathes sp', elapsedTime=PT1H55M26.041S, numOfAssociations=4, numOfImages=0}]}
2022-06-08 19:19:35 [JavaFX Application Thread       ] DEBUG org.mbari.vars.ui.AppController                     AnnotationsSelectedEvent{source=org.mbari.vars.ui.javafx.annotable.AnnotationTableController@7dfa6a82, refs=[Annotation{observationUuid=1d9e230f-46e1-4a60-976b-d55efe75ce1e, concept='Bathypathes sp', elapsedTime=PT1H55M26.041S, numOfAssociations=4, numOfImages=0}]}
2022-06-08 19:19:35 [JavaFX Application Thread       ] DEBUG org.mbari.vars.ui.AppController                     AnnotationsSelectedEvent{source=org.mbari.vars.ui.javafx.annotable.AnnotationTableController@7dfa6a82, refs=[Annotation{observationUuid=1d9e230f-46e1-4a60-976b-d55efe75ce1e, concept='Bathypathes sp', elapsedTime=PT1H55M26.041S, numOfAssociations=4, numOfImages=0}]}
2022-06-08 19:19:35 [JavaFX Application Thread       ] DEBUG org.mbari.vars.ui.AppController                     UpdateAnnotationCmd{oldAnnotation=Annotation{observationUuid=1d9e230f-46e1-4a60-976b-d55efe75ce1e, concept='Bathypathes sp', elapsedTime=PT1H55M26.041S, numOfAssociations=4, numOfImages=0}, newAnnotation=Annotation{observationUuid=1d9e230f-46e1-4a60-976b-d55efe75ce1e, concept='Neomorphaster forcipatus', elapsedTime=PT1H55M26.041S, numOfAssociations=4, numOfImages=0}}
2022-06-08 19:19:35 [org.mbari.vars.ui.commands.CommandManager] DEBUG org.mbari.vars.ui.commands.CommandManager           Executing Command: DO - Update annotation
2022-06-08 19:19:35 [OkHttp [http://hurlstor.soest.hawaii.edu:8082/.](http://hurlstor.soest.hawaii.edu:8082/)..] DEBUG org.mbari.vars.ui.AppController                     AnnotationsChangedEvent{source=null, refs=[Annotation{observationUuid=1d9e230f-46e1-4a60-976b-d55efe75ce1e, concept='Neomorphaster forcipatus', elapsedTime=PT1H55M26.041S, numOfAssociations=4, numOfImages=0}]}
2022-06-08 19:19:35 [OkHttp [http://hurlstor.soest.hawaii.edu:8082/.](http://hurlstor.soest.hawaii.edu:8082/)..] DEBUG org.mbari.vars.ui.AppController                     AnnotationsSelectedEvent{source=org.mbari.vars.ui.javafx.annotable.AnnotationTableController@7dfa6a82, refs=[Annotation{observationUuid=90411b5a-daef-4cfe-7165-ce5afd75ce1e, concept='Neomorphaster forcipatus', elapsedTime=PT1H54M57.898S, numOfAssociations=4, numOfImages=0}]}
2022-06-08 19:19:35 [JavaFX Application Thread       ] DEBUG org.mbari.vars.ui.AppController                     AnnotationsSelectedEvent{source=null, refs=[Annotation{observationUuid=1d9e230f-46e1-4a60-976b-d55efe75ce1e, concept='Neomorphaster forcipatus', elapsedTime=PT1H55M26.041S, numOfAssociations=4, numOfImages=0}]}
2022-06-08 19:19:35 [JavaFX Application Thread       ] DEBUG org.mbari.vars.ui.AppController                     AnnotationsSelectedEvent{source=org.mbari.vars.ui.javafx.annotable.AnnotationTableController@7dfa6a82, refs=[Annotation{observationUuid=1d9e230f-46e1-4a60-976b-d55efe75ce1e, concept='Neomorphaster forcipatus', elapsedTime=PT1H55M26.041S, numOfAssociations=4, numOfImages=0}]}
2022-06-08 19:19:35 [JavaFX Application Thread       ] DEBUG org.mbari.vars.ui.AppController                     AnnotationsSelectedEvent{source=org.mbari.vars.ui.javafx.annotable.AnnotationTableController@7dfa6a82, refs=[Annotation{observationUuid=1d9e230f-46e1-4a60-976b-d55efe75ce1e, concept='Neomorphaster forcipatus', elapsedTime=PT1H55M26.041S, numOfAssociations=4, numOfImages=0}]}
2022-06-08 19:19:35 [JavaFX Application Thread       ] DEBUG o.m.v.ui.javafx.shared.FilteredComboBoxDecorator    Handling KeyCode = ENTER
2022-06-08 19:22:11 [JavaFX Application Thread       ] DEBUG org.mbari.vars.ui.AppController                     org.mbari.vars.ui.messages.UndoMsg@7599e2e
2022-06-08 19:22:11 [org.mbari.vars.ui.commands.CommandManager] DEBUG org.mbari.vars.ui.commands.CommandManager           Executing Command: UNDO - Update annotation
2022-06-08 19:22:11 [OkHttp [http://hurlstor.soest.hawaii.edu:8082/.](http://hurlstor.soest.hawaii.edu:8082/)..] DEBUG org.mbari.vars.ui.AppController                     AnnotationsChangedEvent{source=null, refs=[Annotation{observationUuid=1d9e230f-46e1-4a60-976b-d55efe75ce1e, concept='Bathypathes sp', elapsedTime=PT1H55M26.041S, numOfAssociations=4, numOfImages=0}]}
2022-06-08 19:22:11 [OkHttp [http://hurlstor.soest.hawaii.edu:8082/.](http://hurlstor.soest.hawaii.edu:8082/)..] DEBUG org.mbari.vars.ui.AppController                     AnnotationsSelectedEvent{source=org.mbari.vars.ui.javafx.annotable.AnnotationTableController@7dfa6a82, refs=[Annotation{observationUuid=90411b5a-daef-4cfe-7165-ce5afd75ce1e, concept='Neomorphaster forcipatus', elapsedTime=PT1H54M57.898S, numOfAssociations=4, numOfImages=0}]}
2022-06-08 19:22:11 [JavaFX Application Thread       ] DEBUG org.mbari.vars.ui.AppController                     AnnotationsSelectedEvent{source=null, refs=[Annotation{observationUuid=1d9e230f-46e1-4a60-976b-d55efe75ce1e, concept='Bathypathes sp', elapsedTime=PT1H55M26.041S, numOfAssociations=4, numOfImages=0}]}
2022-06-08 19:22:11 [JavaFX Application Thread       ] DEBUG org.mbari.vars.ui.AppController                     AnnotationsSelectedEvent{source=org.mbari.vars.ui.javafx.annotable.AnnotationTableController@7dfa6a82, refs=[Annotation{observationUuid=1d9e230f-46e1-4a60-976b-d55efe75ce1e, concept='Bathypathes sp', elapsedTime=PT1H55M26.041S, numOfAssociations=4, numOfImages=0}]}
2022-06-08 19:22:11 [JavaFX Application Thread       ] DEBUG org.mbari.vars.ui.AppController                     AnnotationsSelectedEvent{source=org.mbari.vars.ui.javafx.annotable.AnnotationTableController@7dfa6a82, refs=[Annotation{observationUuid=1d9e230f-46e1-4a60-976b-d55efe75ce1e, concept='Bathypathes sp', elapsedTime=PT1H55M26.041S, numOfAssociations=4, numOfImages=0}]}
```

**Comment by hohonuuli** [comment]
Another possible cause: https://vladmihalcea.com/write-skew-2pl-mvcc/

**Comment by SarahRDBingo** [comment]
I was wondering if the concept name change problem is related to the Issue #128 jumpy scroll/selection issue with JavaFX. Previously Meagan mentioned that the name change occurs mostly in tandem with a scroll hop. I've also noticed recently that as I edit a record, sometimes the scroll/selection jump will cause me to make edits to the most recent record as well as the immediately preceding one. I'm wondering if the inconsistent row heights is exacerbating things as well. Could we try maybe forcing a specific row height to see if things change at all? I understand some of the details would be hidden/cut off, but they would still be visible in the editing pane.

**Comment by hohonuuli** [comment]
It might be related to #128. I can't duplicate the issue on my test setup yet, so I can't confirm. I'm waiting on [this pull request](https://github.com/openjdk/jfx/pull/808) to be integrated into JavaFX to see if it addresses the jumping issue.

**Comment by hohonuuli** [comment]
The only place an update to an annotation concept name is triggered is when the concept editor concept box receives an `enter` keystroke. This bug appears to manifest when they are adding associations via a dialog. It's very sporadic. I have not triggered it myself but I've watched it (via screen share) being triggered. Do they hold down the enter key for a long moment?

**Comment by MeaganPutts** [comment]
Interesting... I don't think it has anything to do with holding the key
down for extended periods of time. For myself, I can say that there should
be no difference in my physical execution of the enter key when the bug
appears...

**Comment by SarahRDBingo** [comment]
I wouldn't say the enter key is held for along time, nor is the keyboard sticky. The problem can also occasionally arise when editing a field within the editor concept box. The bug is triggered less when using the custom preset buttons, but it still can happen on a rare occasion.

**Comment by hohonuuli** [comment]
From @MeaganPutts re: vars-annotation 1.4.0-rc5

> This still remains an issue and is worse today than when I was annotating last week.

**Comment by hohonuuli** [comment]
Need feedback on this in for release 1.5.0 🤞

**Comment by MeaganPutts** [comment]
Just downloaded it and will let you know how it goes 👍 
One thing right off the bat (but not a big deal at all) the scroll in the annotation list window is a bit spicy. It takes off when I hover scroll. If there is a way to mellow that out, that would be useful.

**Comment by hohonuuli** [comment]
From @SarahRDBingo re [1.5.0](https://github.com/mbari-media-management/vars-annotation/releases/tag/1.5.0)

>  The concept name change bug seems to be less apparent as well. I'm not sure if that bug was fully fixed with the new listview functions, but I have my fingers crossed.

**Comment by hohonuuli** [comment]
@MeaganPutts Reported that this may be resolved now in (Finally) #156
